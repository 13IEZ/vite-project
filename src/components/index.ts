export * from './Buttons/PrimaryButton/PrimaryButton';
export * from './Modals/ActionsFormModal';
