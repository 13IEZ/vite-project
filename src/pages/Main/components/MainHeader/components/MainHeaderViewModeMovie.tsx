import React from 'react';
import dayjs from 'dayjs';
import { Grid, Container, IconButton, Box } from '@mui/material';
import { ColorEnum, StyledText, StyledImg } from 'style/style';
import SearchIcon from '@mui/icons-material/Search';
import { useTypedSelectorHook, useActions } from 'hooks';
import { IClickedItem } from 'store/types/movieTypes';

export const MainHeaderViewModeMovie = () => {
  const { clickedItem } = useTypedSelectorHook(state => state.movie) as IClickedItem;
  const { resetClickedMovie } = useActions();

  return (
    <header>
      <Container maxWidth='lg'>
        <Grid container justifyContent='space-between' sx={{ margin: '1.25rem 0 3.75rem' }}>
          <StyledText fontSize='125%' fontWeight={300} color={ColorEnum.PINK}>
            <strong>netflix</strong>roulette
          </StyledText>
          <IconButton size='large' color='inherit' onClick={() => resetClickedMovie()}>
            <SearchIcon fontSize='large' />
          </IconButton>
        </Grid>

        <Grid container gap={10} marginBottom='3.75rem' wrap='nowrap'>
          <Grid item xs={4}>
            <StyledImg src={clickedItem?.poster_path} alt='movie poster' />
          </Grid>

          <Grid item xs={8}>
            <Grid container>
              <StyledText fontSize='250%' fontWeight={300} color={ColorEnum.WHITE}>
                {clickedItem?.title}
              </StyledText>
              <Box
                sx={{
                  borderRadius: '50%',
                  border: '0.06rem solid white',
                  margin: 'auto',
                  padding: '0.5rem',
                }}
              >
                <StyledText fontSize='125%' fontWeight={300} color={ColorEnum.WHITE}>
                  {clickedItem?.vote_average}
                </StyledText>
              </Box>
            </Grid>

            <StyledText fontSize='87.5%' fontWeight={300} color={ColorEnum.GREY}>
              {clickedItem?.genres.join(', ')}
            </StyledText>

            <Grid container gap={5} marginTop='1.5rem'>
              <StyledText fontSize='150%' fontWeight={300} color={ColorEnum.PINK}>
                {dayjs(clickedItem?.release_date).year()}
              </StyledText>
              <StyledText fontSize='150%' fontWeight={300} color={ColorEnum.PINK}>
                {clickedItem?.runtime} min
              </StyledText>
            </Grid>

            <StyledText fontSize='125%' fontWeight={300} color={ColorEnum.GREY}>
              {clickedItem?.overview}
            </StyledText>
          </Grid>
        </Grid>
      </Container>
    </header>
  );
};
