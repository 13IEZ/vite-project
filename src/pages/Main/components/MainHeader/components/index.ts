export * from 'pages/Main/components/MainHeader/components/MainHeaderViewModeFilm';
export * from 'pages/Main/components/MainHeader/components/MainHeaderSearchbar';
