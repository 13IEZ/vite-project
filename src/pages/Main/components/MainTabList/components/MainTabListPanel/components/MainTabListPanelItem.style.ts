import styled from '@emotion/styled';
import { Grid, IconButton } from '@mui/material';
import { ColorEnum } from 'style/style';

export const StyledListPanelItem = styled(Grid)`
  transition: transform 0.5s;
  position: relative;

  &:hover {
    transform: scale(1.02);
  }
`;

export const StyledVertButton = styled(IconButton)`
  position: absolute;
  top: 0.5rem;
  right: 0.5rem;
  background-color: ${ColorEnum.DARK};
  opacity: 0.7;
`;
