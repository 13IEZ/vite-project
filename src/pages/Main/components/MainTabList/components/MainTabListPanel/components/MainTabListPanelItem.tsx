import React from 'react';
import { Grid, Menu, MenuItem } from '@mui/material';
import MoreVertIcon from '@mui/icons-material/MoreVert';
import { StyledText, ColorEnum, StyledImg } from 'style/style';
import { StyledListPanelItem, StyledVertButton } from './MainTabListPanelItem.style';
import dayjs from 'dayjs';
import { IMovie, GetMovieType } from 'store/types/movieTypes';
import { useSearchParams } from 'react-router-dom';
import { useActions } from 'hooks';

interface IMainTabListProps {
  item: IMovie;
}

const MainTabListPanelItem: React.FC<IMainTabListProps> = ({ item }) => {
  const { title, poster_path, genres, release_date, id } = item;
  const [anchorEl, setAnchorEl] = React.useState<null | HTMLElement>(null);
  const [searchParams, setSearchParams] = useSearchParams();
  const { deleteMovie, getMovie } = useActions();
  const isOpen = Boolean(anchorEl);

  const handleOpenMenu = (event: React.MouseEvent<HTMLButtonElement>) => {
    setAnchorEl(event.currentTarget);
  };

  const handleCloseMenu = () => {
    setAnchorEl(null);
  };

  const handleOpenModal = (id: number) => {
    searchParams.set('isShowModal', id?.toString() || 'new');
    setSearchParams(searchParams);
    setAnchorEl(null);
  };

  const handleDeleteMovie = (id: number) => {
    deleteMovie(id.toString(), searchParams.toString());
    setAnchorEl(null);
  };

  return (
    <StyledListPanelItem id={'test'}>
      <Grid onClick={() => id && getMovie(id.toString(), GetMovieType.VIEW)}>
        <StyledImg src={poster_path} alt='movie poster' />
      </Grid>

      <Grid
        container
        justifyContent='space-between'
        alignItems='start'
        marginTop='1rem'
        wrap='nowrap'
      >
        <Grid item>
          <StyledText fontSize='112.5%' fontWeight={500} color={ColorEnum.GREY}>
            {title}
          </StyledText>
          <StyledText fontSize='87.5%' fontWeight={500} color={ColorEnum.DARKER_GREY}>
            {genres.join(', ')}
          </StyledText>
        </Grid>
        <Grid
          item
          padding='0.25rem'
          sx={{ border: `0.1rem solid ${ColorEnum.DARKER_GREY}`, borderRadius: '0.25rem' }}
        >
          <StyledText fontSize='87.5%' fontWeight={500} color={ColorEnum.GREY}>
            {dayjs(release_date).year()}
          </StyledText>
        </Grid>
      </Grid>
      <StyledVertButton onClick={handleOpenMenu}>
        <MoreVertIcon color='action' />
      </StyledVertButton>
      <Menu
        id='basic-menu'
        MenuListProps={{
          'aria-labelledby': 'basic-button',
        }}
        anchorEl={anchorEl}
        open={isOpen}
        onClose={handleCloseMenu}
      >
        <MenuItem onClick={() => id && handleDeleteMovie(id)}>Delete</MenuItem>
        <MenuItem onClick={() => id && handleOpenModal(id)}>Edit</MenuItem>
      </Menu>
    </StyledListPanelItem>
  );
};

export default MainTabListPanelItem;
