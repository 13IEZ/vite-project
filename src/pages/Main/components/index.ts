export * from 'pages/Main/components/MainFooter/MainFooter';
export * from 'pages/Main/components/MainHeader/MainHeader';
export * from 'pages/Main/components/MainTabList/MainTabList';
