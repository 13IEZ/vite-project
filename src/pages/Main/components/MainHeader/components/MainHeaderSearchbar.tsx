import React, { useState } from 'react';
import { Grid } from '@mui/material';
import { StyledInput } from 'style/style';
import { PrimaryButton } from 'components';
import { useActions } from 'hooks';
import { useSearchParams } from 'react-router-dom';

export const MainHeaderSearchbar = () => {
  const [value, setValue] = useState('');
  const [searchParams] = useSearchParams();
  const { getMovies } = useActions();

  const handleClickSearch = () => {
    //в апи невозможно искать по названию и сохранить фильтра по жанру
    getMovies(value ? `searchBy=title&search=${value}` : searchParams.toString());
  };

  return (
    <Grid container alignSelf='center' spacing={3} marginTop='2.5rem' paddingBottom='9.375rem'>
      <Grid item xs={9.5}>
        <StyledInput
          value={value}
          onChange={e => setValue(e.target.value)}
          label='What do you want to watch?'
          variant='outlined'
          id='test-id-input'
          fullWidth
        />
      </Grid>
      <Grid item xs={2.5}>
        <PrimaryButton
          clickHandler={handleClickSearch}
          title='search'
          type='contained'
          isFullWidth
          id='test-id-btn'
        />
      </Grid>
    </Grid>
  );
};
