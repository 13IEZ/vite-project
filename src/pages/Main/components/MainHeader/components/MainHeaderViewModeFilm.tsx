import React, { Dispatch, SetStateAction } from 'react';
import dayjs from 'dayjs';
import { Grid, Container, IconButton, Box } from '@mui/material';
import { IIMainTabListPanelItem } from 'pages/Main/components/MainTabList/components/MainTabListPanel/components/MainTabListPanelItem';
import { ColorEnum, StyledText, StyledImg } from 'style/style';
import SearchIcon from '@mui/icons-material/Search';

interface IMainHeaderViewModeFilm {
  clickedItemToView: null | IIMainTabListPanelItem;
  setClickedItemToView: Dispatch<SetStateAction<IIMainTabListPanelItem | null>>;
}

export const MainHeaderViewModeFilm: React.FC<IMainHeaderViewModeFilm> = ({
  clickedItemToView,
  setClickedItemToView,
}) => {
  return (
    <header>
      <Container maxWidth='lg'>
        <Grid container justifyContent='space-between' sx={{ margin: '1.25rem 0 3.75rem' }}>
          <StyledText fontSize='125%' fontWeight={300} color={ColorEnum.PINK}>
            <strong>netflix</strong>roulette
          </StyledText>
          <IconButton size='large' color='inherit' onClick={() => setClickedItemToView(null)}>
            <SearchIcon fontSize='large' />
          </IconButton>
        </Grid>

        <Grid container gap={10} marginBottom='3.75rem' wrap='nowrap'>
          <Grid item xs={4}>
            <StyledImg src={clickedItemToView?.image} alt='movie poster' />
          </Grid>

          <Grid item xs={8}>
            <Grid container>
              <StyledText fontSize='250%' fontWeight={300} color={ColorEnum.WHITE}>
                {clickedItemToView?.title}
              </StyledText>
              <Box
                sx={{
                  borderRadius: '50%',
                  border: '0.06rem solid white',
                  margin: 'auto',
                  padding: '0.5rem',
                }}
              >
                <StyledText fontSize='125%' fontWeight={300} color={ColorEnum.WHITE}>
                  {clickedItemToView?.rating}
                </StyledText>
              </Box>
            </Grid>

            <StyledText fontSize='87.5%' fontWeight={300} color={ColorEnum.GREY}>
              {clickedItemToView?.genre}
            </StyledText>

            <Grid container gap={5} marginTop='1.5rem'>
              <StyledText fontSize='150%' fontWeight={300} color={ColorEnum.PINK}>
                {dayjs(clickedItemToView?.year).year()}
              </StyledText>
              <StyledText fontSize='150%' fontWeight={300} color={ColorEnum.PINK}>
                {clickedItemToView?.runtime} min
              </StyledText>
            </Grid>

            <StyledText fontSize='125%' fontWeight={300} color={ColorEnum.GREY}>
              {clickedItemToView?.overview}
            </StyledText>
          </Grid>
        </Grid>
      </Container>
    </header>
  );
};
