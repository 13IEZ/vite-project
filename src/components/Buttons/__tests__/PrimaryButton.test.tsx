import React from 'react';
import { PrimaryButton } from 'components';
import { expect, it, describe } from 'vitest';
import renderer from 'react-test-renderer';

describe('PrimaryButton', () => {
  it('should be rendered correctly', () => {
    const tree = renderer.create(<PrimaryButton title='Test title' type='outlined' />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
