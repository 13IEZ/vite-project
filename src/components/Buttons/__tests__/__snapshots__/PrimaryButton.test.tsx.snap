// Vitest Snapshot v1

exports[`PrimaryButton > should be rendered correctly 1`] = `
<button
  className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-z5heq2 css-1rwt2y5-MuiButtonBase-root-MuiButton-root"
  disabled={false}
  onBlur={[Function]}
  onContextMenu={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  Test title
</button>
`;
